libbeat/fields/rollup/rollup.go

```diff
@@ -0,0 +1,81 @@
+package rollup
+
+import (
+	"github.com/elastic/beats/libbeat/common"
+)
+
+// Processor struct to process fields to rollup
+type Processor struct {
+	metrics []common.MapStr
+	terms   []string
+}
+
+// NewProcessor creates a new rollup fields processor. It presets the required terms for Metricbeat.
+func NewProcessor() *Processor {
+	return &Processor{
+		terms: []string{"beat.name", "metricset.name", "metricset.module"},
+	}
+}
+
+// Process recursively processes the given fields and checks for rollup configs
+func (p *Processor) Process(fields common.Fields, path string) error {
+	for _, field := range fields {
+
+		field.Path = path
+
+		switch field.Type {
+		case "group":
+			var newPath string
+			if path == "" {
+				newPath = field.Name
+			} else {
+				newPath = path + "." + field.Name
+			}
+
+			if err := p.Process(field.Fields, newPath); err != nil {
+				return err
+			}
+		default:
+			if len(field.RollupMetrics) > 0 {
+				p.addMetric(field)
+			}
+			if field.RollupTerm {
+				p.addTerm(field)
+			}
+		}
+	}
+	return nil
+}
+
+func (p *Processor) addMetric(f common.Field) {
+	metric := common.MapStr{
+		"field":   f.Path + "." + f.Name,
+		"metrics": f.RollupMetrics,
+	}
+	p.metrics = append(p.metrics, metric)
+}
+
+func (p *Processor) addTerm(f common.Field) {
+	p.terms = append(p.terms, f.Path)
+}
+
+// Generate creates a common.MapStr object for the rollup job with the terms and fields.
+func (p *Processor) Generate() common.MapStr {
+	return common.MapStr{
+		"index_pattern": "metricbeat-*",
+		"rollup_index":  "metricbeat_rollup",
```

Comment:
@polyfractal Any recommendations on default naming here. I remember that metricbeat-* prefix is probably not the best idea as it matches the same pattern as the default indices.

Reply:
`metricbeat_rollup` seems reasonable.  Or maybe `rollup_metricbeat` or `rollup-metricbeat` just so it's less likely for someone to confuse the two with an index pattern?  I don't think we've really established a convention yet so this is the first :)

You're right that `metricbeat-rollup` (with the hyphen) would match itself and throw an exception in 6.4.

Reply:
My favorite would have been `metricbeat-rollup` as it makes it clear it's almost the same data but I understand that this breaks things.

I'm probably going to stick to `rollup-metricbeat-{beat.version}` for now as we use `-` everywhere else for indices.